src/app/modules/time-entries/pages/time-entries.component.ts

```diff
@@ -101,11 +101,11 @@ export class TimeEntriesComponent implements OnInit, OnDestroy {
 
   saveEntry(event: SaveEntryEvent): void {
     if (this.activeTimeEntry !== null && this.activeTimeEntry !== undefined) {
-      const entryDateAsIso = new Date(event.entry.start_date).toISOString();
-      const entryDateAsLocalDate = new Date(entryDateAsIso);
+      const startDateAsLocalDate = new Date(event.entry.start_date);
+      const endDateAsLocalDate = new Date(event.entry.end_date);
       const activeEntryAsLocalDate = new Date(this.activeTimeEntry.start_date);
       const isEditingActiveEntry = this.entryId === this.activeTimeEntry.id;
-      if (!isEditingActiveEntry && entryDateAsLocalDate > activeEntryAsLocalDate) {
+      if (!isEditingActiveEntry && (startDateAsLocalDate > activeEntryAsLocalDate || endDateAsLocalDate > activeEntryAsLocalDate)) {
```

Comment:
Don't hesitate to use more variables instead of using operators inside the conditions. For example:
`isStartDateGreaterThanActiveEntry = startDateAsLocalDate > activeEntryAsLocalDate
isEndDateGreaterThanActiveEntry = endDateAsLocalDate > activeEntryAsLocalDate`
Then the condition would be more readable. 

Besides, due to the previous comment, about changing the order of the statement. The result would be something like this:

```
isStartDateLowerThanActiveEntry = startDateAsLocalDate < activeEntryAsLocalDate
isEndDateLowerThanActiveEntry = endDateAsLocalDate < activeEntryAsLocalDate
if (isEditingActiveEntry && (isStartDateLowerThanActiveEntry || isEndDateLowerThanActiveEntry)) {
  this.doSave(event);
} 
else {
  this.toastrService.error('You are on the clock and this entry overlaps it, try with earlier times.');
}
```